src/lib/rnp.cpp

```diff
@@ -2096,11 +2096,12 @@ FFI_GUARD
 rnp_result_t
 rnp_input_from_memory(rnp_input_t *input, const uint8_t buf[], size_t buf_len, bool do_copy)
 try {
-    if (!input || !buf) {
+    if (!input || (!buf && buf_len > 0)) {
```

Comment:
It would be logical to replicate behaviour of `init_mem_src()` here, which allows null buffer for the zero-size input.